proxy/grpcproxy/cache/store.go

```diff
@@ -73,7 +77,11 @@ func (c *cache) Add(req *pb.RangeRequest, resp *pb.RangeResponse) {
 	defer c.mu.Unlock()
 
 	if req.Revision > c.compactedRev {
-		c.lru.Add(key, resp)
+		if req.Revision == 0 {
+			c.lru.Set(key, resp, defaultCurrentTTL)
+		} else {
+			c.lru.Set(key, resp, defaultHistoricTTL)
+		}
```

Comment:
@fanminshi The cache is keyed by the request data. Get(rev=100) will always return rev=100; it is not subject to change (ignoring, rev=100 doesn't exist yet or rev=100 is compacted away, which is already checked). Get(rev=0) will not always return rev=100; it is subject to change.